### PR TITLE
IGNITE-19766 Added topology snapshot info to services system view

### PR DESCRIPTION
**Pull request by NSAmelchev**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/spi/systemview/view/ServiceView.java [hunk 1]
```diff
@@ -98,4 +99,14 @@ public boolean staticallyConfigured() {
     public UUID originNodeId() {
         return serviceInfo.originNodeId();
     }
+
+    /**
+     * Gets service deployment topology snapshot. Service topology snapshot is represented
```
Comment (Vladsz83) [inline modules/core/src/main/java/org/apache/ignite/spi/systemview/view/ServiceView.java:104]: Other descriptions of the properties  are short int this class. Maybe just '@return Map of number of service instances per node ID.' and '@see ServiceDescriptor'? WDYT?

Comment (NSAmelchev) [reply modules/core/src/main/java/org/apache/ignite/spi/systemview/view/ServiceView.java]: Done

File: modules/core/src/test/java/org/apache/ignite/internal/metric/SystemViewSelfTest.java [hunk 2]
```diff
@@ -289,6 +289,7 @@ public void testServices() throws Exception {
                 assertEquals(TestNodeFilter.class, sview.nodeFilter());
                 assertFalse(sview.staticallyConfigured());
                 assertEquals(g.localNode().id(), sview.originNodeId());
+                assertEquals(F.asMap(g.localNode().id(), 1), sview.topologySnapshot());
```
Comment (Vladsz83) [inline modules/core/src/test/java/org/apache/ignite/internal/metric/SystemViewSelfTest.java:291]: Btw. Can we also remove repeated 'assertEquals(srvcCfg.getMaxPerNodeCount(), sview.maxPerNodeCount());' above ?

Comment (NSAmelchev) [reply modules/core/src/test/java/org/apache/ignite/internal/metric/SystemViewSelfTest.java:291]: Done

File: modules/control-utility/src/test/java/org/apache/ignite/util/SystemViewCommandTest.java [hunk 3]
```diff
@@ -346,6 +346,7 @@ public void testServices() {
         assertEquals(srvcCfg.getName(), sysView.get(1)); // name
         assertEquals(DummyService.class.getName(), sysView.get(2)); // serviceClass
         assertEquals(Integer.toString(srvcCfg.getMaxPerNodeCount()), sysView.get(6)); // maxPerNodeCount
+        assertEquals(F.first(ignite0.services().serviceDescriptors()).topologySnapshot().toString(), sysView.get(10));
```
Comment (Vladsz83) [inline modules/control-utility/src/test/java/org/apache/ignite/util/SystemViewCommandTest.java:349]: Suggestion:
`for(Ignite ig : G.allGrids()){
            List<List<String>> srvsView = systemView((IgniteEx)ig, SVCS_VIEW);

            assertEquals(1, srvsView.size());

            List<String> sysView = srvsView.get(0);

            assertEquals(srvcCfg.getName(), sysView.get(1)); // name
            assertEquals(DummyService.class.getName(), sysView.get(2)); // serviceClass
            assertEquals(Integer.toString(srvcCfg.getMaxPerNodeCount()), sysView.get(6)); // maxPerNodeCount
            assertEquals(F.first(ig.services().serviceDescriptors()).topologySnapshot().toString(), sysView.get(10));
        }`
Up2U

Comment (NSAmelchev) [reply modules/control-utility/src/test/java/org/apache/ignite/util/SystemViewCommandTest.java:349]: Removed the topology check on a second node

File: modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/metric/SqlViewExporterSpiTest.java [hunk 4]
```diff
@@ -309,6 +310,7 @@ public void testServices() {
         assertEquals(srvcCfg.getName(), sysView.get(0));
         assertEquals(DummyService.class.getName(), sysView.get(2));
         assertEquals(srvcCfg.getMaxPerNodeCount(), sysView.get(4));
+        assertEquals(F.first(ignite0.services().serviceDescriptors()).topologySnapshot().toString(), sysView.get(10));
```
Comment (Vladsz83) [inline modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/metric/SqlViewExporterSpiTest.java:313]: Can we check the same in `JmxExporterSpiTest#testServices()`. Does it work? Seems a type conversion error occurs there.

File: modules/core/src/main/java/org/apache/ignite/internal/managers/systemview/SystemViewMBean.java [hunk 5]
```diff
@@ -289,7 +290,7 @@ else if (clazz.isEnum())
             else if (clazz.isAssignableFrom(Class.class))
                 data.put(name, ((Class<?>)val).getName());
             else if (clazz.isAssignableFrom(IgniteUuid.class) || clazz.isAssignableFrom(UUID.class) ||
-                clazz.isAssignableFrom(InetSocketAddress.class))
+                clazz.isAssignableFrom(InetSocketAddress.class) || clazz.isAssignableFrom(AbstractMap.class))
```
Comment (Vladsz83) [inline modules/core/src/main/java/org/apache/ignite/internal/managers/systemview/SystemViewMBean.java:293]: clazz.isAssignableFrom(Map.class) ?

File: modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/metric/SqlViewExporterSpiTest.java (hunk 4 above)
Comment (Vladsz83) [inline modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/metric/SqlViewExporterSpiTest.java:313]: Up2U: we might also add `assertEquals(F.asMap(ignite.localNode().id(), 1).toString(), sysView.get("topologySnapshot"));` to `JmxExporterSpiTest#testServices()`

Comment (NSAmelchev) [reply modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/metric/SqlViewExporterSpiTest.java:313]: Done

**Review by Vladsz83 — APPROVED** [review]

**Pull request merged**